### PR TITLE
fix(rlpTxn): constraint ADDR during AccessList tuple & ChainId

### PR DESCRIPTION
**Pull request by letypequividelespoubelles**
spec : https://github.com/Consensys/linea-specification-internal/pull/620 & https://github.com/Consensys/linea-specification-internal/pull/618

### CODE REVIEW

File: rlptxn/constraints.lisp [hunk 1]
```diff
@@ -829,10 +829,9 @@
 
 ;; 4.5.2.15
 (defconstraint phaseAccessList-updateAddrLookUp (:guard IS_PHASE_ACCESS_LIST)
-  (if-zero (+ [DEPTH 2]
-              (- (prev nADDR) nADDR))
-           (begin (remained-constant! ADDR_HI)
-                  (remained-constant! ADDR_LO))))
+  (if-not-zero (* [DEPTH 1]  (- nADDR (- (prev nADDR) 1)))
+                     (begin (remained-constant! ADDR_HI)
+                            (remained-constant! ADDR_LO))))
```
Comment (OlivierBBB) [inline rlptxn/constraints.lisp:834]: I'm not following how this maps to

image

Comment (OlivierBBB) [reply rlptxn/constraints.lisp:834]: I guess this relates to this spec update: https://github.com/Consensys/linea-specification-internal/pull/618

Comment (OlivierBBB) [reply rlptxn/constraints.lisp:834]: My next question would be: are special precautions required for the first row of `IS_PHASE_ACCESS_LIST = 1` ?

Comment (letypequividelespoubelles) [reply rlptxn/constraints.lisp:834]: No, the first row of ACCESS£LIST PHASE  has DEPTH_1 = 0, so it's OK.

Comment (letypequividelespoubelles) [reply rlptxn/constraints.lisp:834]: see

File: rlptxn/constraints.lisp [hunk 2]
```diff
@@ -858,7 +864,9 @@
                                                 (begin (vanishes! (+ PHASE_END
                                                                      (next LT)
                                                                      (- 1 (next LX))
-                                                                     (- 1 (next IS_PREFIX)))))))))))
+                                                                     (- 1 (next IS_PREFIX))))
+                                                       (eq! (^ (w-minus-two-beta-minus-protected-base-V) 2)
+                                                            (w-minus-two-beta-minus-protected-base-V)))))))))
```
Comment (OlivierBBB) [inline rlptxn/constraints.lisp:869]: We could just use `(is-binary (w-minus-two-beta-minus-protected-base-V))`, no ?

**Review by OlivierBBB — APPROVED** [review]

File: rlptxn/constraints.lisp (hunk 2 above)
Comment (letypequividelespoubelles) [reply rlptxn/constraints.lisp:869]: yes, but I plan to do it globally as I have this perf issue in lot of space in RLP_* module : https://github.com/Consensys/linea-constraints/issues/396

Comment (letypequividelespoubelles) [reply rlptxn/constraints.lisp:869]: hum ... c'est pas faux

**Pull request merged**